bot/cogs/doc.py

```diff
@@ -0,0 +1,546 @@
+import asyncio
+import functools
+import logging
+import random
+import re
+from collections import OrderedDict
+from ssl import CertificateError
+from typing import Dict, List, Optional, Tuple
+
+import discord
+from aiohttp import ClientConnectorError
+from bs4 import BeautifulSoup
+from discord.ext import commands
+from markdownify import MarkdownConverter
+from requests import ConnectionError
+from sphinx.ext import intersphinx
+
+from bot.constants import (
+    ADMIN_ROLE, ERROR_REPLIES, MODERATOR_ROLE,
+    OWNER_ROLE, SITE_API_KEY, SITE_API_URL
+)
+from bot.decorators import with_role
+from bot.exceptions import CogBadArgument
+
+
+log = logging.getLogger(__name__)
+logging.getLogger('urllib3').setLevel(logging.WARNING)
+
+
+UNWANTED_SIGNATURE_SYMBOLS = ('[source]', '¶')
+WHITESPACE_AFTER_NEWLINES_RE = re.compile(r"(?<=\n\n)( +)")
+
+
+def async_cache(max_size=128, arg_offset=0):
+    """
+    LRU cache implementation for coroutines.
+
+    :param max_size:
+    Specifies the maximum size the cache should have.
+    Once it exceeds the maximum size, keys are deleted in FIFO order.
+    :param arg_offset:
+    The offset that should be applied to the coroutine's arguments
+    when creating the cache key. Defaults to `0`.
+    """
+
+    # Assign the cache to the function itself so we can clear it from outside.
+    async_cache.cache = OrderedDict()
```

Comment:
Don't set it to the function, make it a local variable instead

Reply:
The entire point of setting it to the function is so we can clear the cache from the outside of it when necessary. Did you read the comment above?